temba/api/v2/views.py

```diff
@@ -3573,3 +3581,454 @@ def post(self, request, *args, **kwargs):
+            response = requests.delete(url, headers=self.parse_headers)
+
+            if response.status_code == 200:
+                org.remove_collection_from_org(user=self.request.user, index=collection_index, collection_type=LOOKUPS)
+                return Response(status=status.HTTP_204_NO_CONTENT)
+            else:
+                return Response(response.json(), status=response.status_code)
+        else:
+            return Response(response_purge.json(), status=response_purge.status_code)
+
+    def put(self, request, *args, **kwargs):
+        org, collection_name, collections_list, error_response = self.get_default_params(config="collection_exist")
+        if error_response:
+            return error_response
+
+        fields_to_skip = ["objectId", "createdAt", "updatedAt", "ACL"]
+        fields_to_create = dict(
+            filter(lambda x: x[0] not in fields_to_skip, self.request.data.get("fields", {}).items())
+        )
+        items_to_push = self.request.data.get("items", [])
+        if not fields_to_create and not items_to_push:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+
+        collection = self.get_collection_full_name(org, collection_name)
+
+        # clear previous data
+        parse_url = f"{settings.PARSE_URL}/schemas/{collection}"
+        response = requests.get(parse_url, headers=self.parse_headers)
+        if response.status_code == 200 and "fields" in response.json():
+            fields = response.json().get("fields")
+
+            for key in list(fields.keys()):
+                if key in ["objectId", "updatedAt", "createdAt", "ACL"]:
+                    del fields[key]
+                else:
+                    del fields[key]["type"]
+                    fields[key]["__op"] = "Delete"
+
+            remove_fields = {"className": collection, "fields": fields}
+
+            purge_url = f"{settings.PARSE_URL}/purge/{collection}"
+            response_purge = requests.delete(purge_url, headers=self.parse_headers)
+
+            if response_purge.status_code in [200, 404]:
+                requests.put(parse_url, data=json.dumps(remove_fields), headers=self.parse_headers)
+
+        # create new columns
+        if fields_to_create:
+            response = requests.put(
+                parse_url,
+                data=json.dumps({"className": collection, "fields": fields_to_create}),
+                headers=self.parse_headers,
+            )
+            if response.status_code != 200:
+                return Response(response.json(), status=response.status_code)
+
+        # insert data rows
+        if items_to_push:
+            requests_ = []
+            insert_url = f"{settings.PARSE_URL}/batch"
+            db_endpoint = f"{settings.PARSE_ENDPOINT}/classes/{collection}"
+            for index, data in enumerate(items_to_push):
+                requests_.append({"method": "POST", "path": db_endpoint, "body": {"order": index, **data}})
```

Comment:
In theory, it would work perfectly, but we have some clients that would like to upload a bunch of rows like 10k, 50k, 100k... So in this case, our Parse server would receive a big request to handle, which would not be a good approach in terms of scale since this number of items could be increased depending on the client. So can we make it the same as on the Parse upload process via GUI? I mean, using bulk insertion but with a limited number of rows per request like here https://github.com/greatnonprofits-nfp/rapidpro/blob/release/2.7/temba/orgs/tasks.py#L192